docs/default.nix

```diff
@@ -36,9 +41,40 @@
       docs_url = "https://developer.gnome.org/gobject/stable";
     };
   };
+
+  lib = name: namespace: description: {
```

Comment:
Why not name this `docBuilder`?

Reply:
From my initial message:

> The documentation derivation still needs more cleaning (lib is a terrible name to give to a helper function) but I ran out of time for today.

I intend to redo that one as well (mainly for arguments, because I hate foo: bar: baz: for multiple arguments) but that will have to be for another time.